src/csdap_bulk_download/client.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Literal
+from urllib.parse import parse_qs, urlparse
+
+from requests import Response, Session
+from requests.auth import AuthBase, HTTPBasicAuth
+
+from .exceptions import AuthError
+
+logger = logging.getLogger(__name__)
+
+METHOD = Literal["GET"] | Literal["POST"]
+
+API_URL_STAGING = "https://csdap-staging.ds.io/api"
+API_URL_PRODUCTION = "https://csdap.earthdata.nasa.gov/api"
```

Comment:
I think we might want to prune the `api` from the end here and tack on `api` for the relevant service interactions (auth & downloads)? I'm thinking ahead to the interactions with STAC search, which don't require auth but might be reasonable to base off of `urljoin(self.api_url, "stac/search")`